docs/root/intro/arch_overview/matching/matching_api.rst

```diff
@@ -0,0 +1,112 @@
+.. _arch_overview_matching_api:
+
+Matching API
+============
+
+Envoy makes use of a matching API to allow the various subsystems to express actions that should be performed based on incoming data.
+
+The matching API is designed as a tree structure to allow for sublinear matching algorithms, and make heavy use of extension points to make it
+easy to extend to different inputs based on protocol or environment data, sublinear matchers and direct matchers. 
+
+Within supported environments (currently only HTTP filters), a wrapper proto can be used to instantiate a matching filter associated with the
+wrapped structure:
+
+.. code-block:: yaml
```

Comment:
@phlax should this be a verified code example? Can we generally just enable this for all new code fragments?

Reply:
the code that i have added verification for so far were complete configurations which are then added to the rst with `literalinclude`

we can do this here and validate just the yaml, or we could wrap this example in a complete config and just highlight the important parts

Reply:
Do I get validation for free just by following the literalinclude pattern used elsewhere? Or do I need to do anything else?

It's easy enough to include enough config to make it a valid config, so if we can easily validate the config by doing that then I think it's worth it

Reply:
its kinda the other way round in a sense

if you separate the config it will automatically get validated unless its specifically excluded

Reply:
...at least for the envoy config - not sure about any yaml validation

Reply:
> Do I get validation for free just by following the literalinclude pattern used elsewhere? Or do I need to do anything else?

so, yep